Flaky test: TestRunContainerWithRmFlagCannotStartContainer (on windows)

This test has failed frequently recently; https://github.com/moby/moby/search?q=TestRunContainerWithRmFlagCannotStartContainer&type=issues

- was once marked as racy/flaky a long time ago: https://github.com/moby/moby/issues/11966
- originally fixed in: https://github.com/moby/moby/pull/12206

```
=== FAIL: github.com/docker/docker/integration-cli TestDockerSuite/TestRunContainerWithRmFlagCannotStartContainer (2.17s)
    docker_cli_run_test.go:2770: Expected not to have containers 357547ca53b5
        
    --- FAIL: TestDockerSuite/TestRunContainerWithRmFlagCannotStartContainer (2.17s)

=== FAIL: github.com/docker/docker/integration-cli TestDockerSuite (3440.51s)
```

--- a/integration-cli/docker_cli_run_test.go
+++ b/integration-cli/docker_cli_run_test.go
@@ -2745,30 +2745,29 @@ func (s *DockerSuite) TestRunVolumesFromRestartAfterRemoved(c *testing.T) {
 
 // run container with --rm should remove container if exit code != 0
 func (s *DockerSuite) TestRunContainerWithRmFlagExitCodeNotEqualToZero(c *testing.T) {
-	existingContainers := ExistingContainerIDs(c)
 	name := "flowers"
 	cli.Docker(cli.Args("run", "--name", name, "--rm", "busybox", "ls", "/notexists")).Assert(c, icmd.Expected{
 		ExitCode: 1,
 	})
 
-	out := cli.DockerCmd(c, "ps", "-q", "-a").Combined()
-	out = RemoveOutputForExistingElements(out, existingContainers)
-	if out != "" {
-		c.Fatal("Expected not to have containers", out)
-	}
+	cli.Docker(cli.Args("container", "inspect", name)).Assert(c, icmd.Expected{
+		ExitCode: 1,
+		Out:      "[]\n",
+		Err:      "o such container", // (N|n)o such container
+	})
 }
 
 func (s *DockerSuite) TestRunContainerWithRmFlagCannotStartContainer(c *testing.T) {
-	existingContainers := ExistingContainerIDs(c)
 	name := "sparkles"
 	cli.Docker(cli.Args("run", "--name", name, "--rm", "busybox", "commandNotFound")).Assert(c, icmd.Expected{
 		ExitCode: 127,
 	})
-	out := cli.DockerCmd(c, "ps", "-q", "-a").Combined()
-	out = RemoveOutputForExistingElements(out, existingContainers)
-	if out != "" {
-		c.Fatal("Expected not to have containers", out)
-	}
+
+	cli.Docker(cli.Args("container", "inspect", name)).Assert(c, icmd.Expected{
+		ExitCode: 1,
+		Out:      "[]\n",
+		Err:      "o such container", // (N|n)o such container
+	})
 }
 
 func (s *DockerSuite) TestRunPIDHostWithChildIsKillable(c *testing.T) {

